hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/impl/connector/kafka/StreamKafkaP.java

```diff
@@ -104,6 +115,15 @@ private void assignPartitions(boolean seekToBeginning) {
                 consumer.seekToBeginning(newAssignments);
             }
         }
+        // initialize offsets
+        for (int topicIdx = 0; topicIdx < partitionCounts.size(); topicIdx++) {
+            int count = partitionCounts.get(topicIdx);
+            offsets.merge(topics.get(topicIdx), LongStream.generate(() -> -1).limit(count).toArray(),
```

Comment:
would appreciate a small comment here, took me a little time to clarify what it was doing (copying old offsets over)